.github/workflows/build-deploy-cloudrun-function.yml

```diff
@@ -0,0 +1,111 @@
+
+# Trigger the workflow on push events to the main branch
+on:
+  push:
+    branches:
+      - $default-branch
+      - staging
+  workflow_call:
+    # Allows manual build and deploy of any branch/ref
+    inputs:
+      function_name:
+        description: 'The name of the Cloud Function to deploy'
+        type: string
+        required: true
+    secrets:
+      WORKLOAD_IDENTITY_PROVIDER:
+        description: 'Workload Identity Provider'
+        required: true
+      GCP_SERVICE_ACCOUNT:
+        description: 'GCP Service Account Name'
+        required: true
+      GCP_SERVICE_ACCOUNT_EMAIL:
+        description: 'GCP Service Account Email'
+        required: true
+      RUN_SERVICE_ACCOUNT:
+        description: 'Run Service Account'
+        required: true
+      GCP_PROJECT_ID:
+        description: 'GCP Project ID'
+        required: true
+      GCP_REGION:
+        description: 'GCP Region'
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      build-number: ${{ steps.build-number.outputs.build-number }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Extract Commit Hash
+        id: build-number
+        run: echo "::set-output name=build-number::$(git rev-parse --short=7 ${{ github.sha }})"
+
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    # Set permissions for the job
+    permissions:
+      contents: read
+      id-token: write
+      pull-requests: write
+
+    steps:
+      # Step 1: Checkout the code from the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Step 2: Authenticate to Google Cloud
+      - id: 'auth'
+        name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: '${{ secrets.GCP_SERVICE_ACCOUNT }}@${{ secrets.GCP_PROJECT_ID }}.iam.gserviceaccount.com'
+
+      # Step 3: Set up the Google Cloud SDK
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v2
+
+      # Step 4: Deploy the Cloud Function using gcloud
+      - name: Deploy to Cloud Run Functions
+        id: deploy
+        env:
+          GCP_REGION: ${{ secrets.GCP_REGION }}
+        run: |
+          gcloud config set project ${{ secrets.GCP_PROJECT_ID }}
+          gcloud functions deploy ${{ inputs.function_name }}-${{ needs.build.outputs.build-number }} \
+           --no-allow-unauthenticated \
+            --region $GCP_REGION \
+            --gen2 \
+            --source=./${{ inputs.function_name }} \
+            --runtime=python312 \
+            --entry-point=hello_http \
+            --trigger-http \
+            --set-secrets=API_KEY=${{ inputs.function_name }}_API_KEY:latest,API_SECRET=${{ inputs.function_name }}_API_SECRET:latest \
```

Comment:
I would have hoped this line wouldn't be necessary, because it's set up in terraform.

Reply:
Where in Terraform is this set? I'm assuming you mean all four lines 88,89,90,91.

Reply:
not 88-90; just 91.
It's here: https://github.com/CruGlobal/cru-terraform-modules/blob/cloudrun/gcp/cloudrun-function/scheduled-tasks/main.tf#L72-L80